pilot/pkg/xds/workload.go

```diff
@@ -51,20 +51,28 @@ func (e WorkloadGenerator) GenerateDeltas(
 		// Nothing changed..
 		return nil, nil, model.XdsLogDetails{}, false, nil
 	}
+
 	subs := sets.New(w.ResourceNames...)
-	addresses := updatedAddresses
-	// If it is not a wildcard, filter out resources we are not subscribed to
-	if !w.Wildcard {
-		addresses = addresses.Intersection(subs)
+	var addresses sets.String
+	if isReq {
+		// this is from request, we only send response for the subscribed address
+		// At t0, a client request A, we only send A and aditional resources back to the client.
+		// At t1, a client request B, we only send B and aditional resources back to the client, no A here.
+		addresses = req.Delta.Subscribed
```

Comment:
Seems like we are ignoring req.Delta.Unsubscribed?

Reply:
We already removed the `Unsubscribed` when calculate `subs` in `deltaWatchedResources`.

So here we donot need to remove again